backend/app/services/project_service.py

```diff
@@ -47,3 +49,33 @@ def delete_project_by_id(self, project_id: UUID) -> None:
                 raise ResourceInUseError(ResourceType.PROJECT, str(project_id), MSG_ERR_DELETE_ACTIVE_PROJECT)
             self._persistence.delete_by_id(project_id, db)
             db.commit()
+
+    def get_project_thumbnail_path(self, project_id: UUID) -> Path | None:
+        """Get the path to the project's thumbnail image, selecting one if none is set"""
+        project = self.get_project_by_id(project_id)
+
+        if project.thumbnail_id:
+            thumbnail_path = self._get_thumbnail_path_for_item(project_id=project_id, thumbnail_id=project.thumbnail_id)
+            if thumbnail_path.exists():
+                return thumbnail_path
+        else:
+            with get_db_session() as db:
+                project_repo = ProjectRepository(db)
+                dataset_repo = DatasetItemRepository(str(project_id), db)
+
+                dataset_items = dataset_repo.list_items(limit=10, offset=0)
+                for item in dataset_items:
+                    thumbnail_path = self._get_thumbnail_path_for_item(project_id=project_id, thumbnail_id=item.id)
+                    if thumbnail_path.exists():
```

Comment:
Why do you need to check if the path exists? Every dataset item is guaranteed to have a thumbnail, no?